server/gateway.go

```diff
@@ -2098,7 +2107,7 @@ func (c *client) processInboundGatewayMsg(msg []byte) {
 	}
 
 	// Check to see if we need to map/route to another account.
-	if acc.imports.services != nil {
+	if acc.imports.services != nil && isServiceReply(c.pa.subject) {
```

Comment:
Maybe add a comment above just stating that we are handling an response to a request that we mapped via service imports, so if we are here we are the origin server.

Reply:
ok